content/blog/2020-07-15-teaching-the-tidyverse-in-2020-part-2-data-visualisation/index.Rmarkdown

```diff
@@ -0,0 +1,162 @@
+  geom_col() +
+  scale_x_continuous(labels = label_percent(accuracy = 1)) +
+  labs(
+    x = "Percentage",
+    y = "Species",
+    title = "Species distribution of penguins",
+    subtitle = "Adelie, Gentoo, and Chinstrap Penguins at Palmer Station LTER",
+    caption = "Source: allisonhorst.github.io/palmerpenguins"
+  )
+```
+
+Another very attractive feature of ggplot2 is that you get legends for free. But sometimes the factor levels displayed in the legends may not be optimally ordered. In these cases `fct_reorder2()` might be helpful. See [this gist](https://gist.github.com/jennybc/847c6b43c4e35cec2e5bb30a3f38af73) from Jenny Bryan for an example using this function to reorder levels displayed on a legend of a line plot.
+
+ggplot2 is a vast package, and I don't think it's feasible to teach the complete functionality of the package within an intro stats or data science curriculum. That being said, I think it is important to not just teach how to make plots, but how to make good plots that follow data visualisation best practices. Leveraging packages like forcats and scales allows you to do that without adding too much cognitive burden on the learners.
+
+## The rest of the gg-verse
+
+There are many packages that extend ggplot2 (e.g. [gganimate](https://gganimate.com/), [ggridges](https://wilkelab.org/ggridges/), [ggrepel](https://ggrepel.slowkow.com/), [GGally](https://ggobi.github.io/ggally/)) and just like it's not possible to cover all of ggplot2 in an intro course, it's definitely not possible to cover the entire gg-verse. However it is useful to introduce your students to some of these packages just so they are aware of the existence of the larger ecosystem.
+
+A particularly good place to start is when you find a pain point with ggplot2 and the solution is using one of these other packages (e.g. ggrepel for labelling data points on a plot without overlapping text).
+
+Another good approach for introducing these packages are assignments where students independently explore packages you haven't explicitly taught them. I like doing this using the following two assignments:
+
+-   **You gotta pick a package or two:** Provide a list of fun and/or easy to onboard packages and ask students to read the package documentation and use the package to do "something".
+
+-   **Mirror, mirror on the wall, who's the ugliest of them all?** I got the idea for this assignment Give a simple plot and ask students to make the plot as ugly as possible. I like doing this with a disclaimer that says "*I realize that ugly is subjective, so we're mostly looking to see if you can figure out how to change the look of a plot using help files of functions you haven't learned before. As you do, try to make your plot as visually unappealing as possible.*"I got the idea for this assignment from Drew Steen, who shared [on Twitter](https://twitter.com/biogeobiochem/status/1172547837046820864) sample student plots that are, well, awful, in the best way possible.
```

Comment:
```suggestion
-   **Mirror, mirror on the wall, who's the ugliest of them all?** Give a simple plot and ask students to make the plot as ugly as possible. I like doing this with a disclaimer that says "*I realize that ugly is subjective, so we're mostly looking to see if you can figure out how to change the look of a plot using help files of functions you haven't learned before. As you do, try to make your plot as visually unappealing as possible.*"I got the idea for this assignment from Drew Steen, who shared [on Twitter](https://twitter.com/biogeobiochem/status/1172547837046820864) sample student plots that are, well, awful, in the best way possible.
```

Reply:
Allison Horst also has a great blog post on using this for teaching: https://www.allisonhorst.com/post/do-your-worst/

Reply:
Thanks!